### PR TITLE
Apps entity restrictions

### PR DESCRIPTION
**Pull request by amitaibu**

### CODE REVIEW

File: app/_posts/2015-12-15-apps-entity-restrictions.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+## Restful Integration and Other bonuses
+
+There is a cool and easy restful integration. The module provides a set of
+[API](https://github.com/RoySegall/apps_entity_restrictions#restful-integration)
+for developers to get this restriction validation on their endpoint. If you are dealing with a decoupled Drupal, you should probably look at this.
+
+The next step is baking more statistics in. Wouldn't it be great if you could know the usages for each
+application? Knowing the amount of requests, failed and passed, is a great tool
+and could detect tryings for hijacking information from un-authorized source.
```
Comment (amitaibu) [inline app/_posts/2015-12-15-apps-entity-restrictions.md]: @RoySegall can you add an image here - even if still doesn't fully working - even a vision of what you want.

File: app/_posts/2015-12-15-apps-entity-restrictions.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+as an enthusiastic Drupal developer willing to solve any Drupal issue (and you can assume they had a few). My part was to write the backend, and the obvious choice was using Restful module.
+
+Then I came across a big problem: How can we manage 3rd party applications and make
+sure they can't access resources which they shouldn't have access to?
+How can we prevent from the Secret Santa application (and app that provides addresses of other
+Burning Man attendees so they could receive gifts) to access the user's medical
+qualification data?
+
+<div class="thumbnail">
+  <img src="{{BASE_PATH}}/assets/images/posts/apps-entity-restrictions/aer.gif">
```
Comment (amitaibu) [inline app/_posts/2015-12-15-apps-entity-restrictions.md]: @RoySegall you can keep the gif as backup - but please attach here a gyfcat versnio

Comment (amitaibu) [inline app/_posts/2015-12-15-apps-entity-restrictions.md]: @RoySegall נו? :wink:

**Comment by RoySegall** [comment]
@amitaibu I can't push my fixes to this one since I don't have push access to this repo.

**Comment by amitaibu** [comment]
no, you should have access - can you please confirm

**Comment by RoySegall** [comment]
@amitaibu I pushed the new changes.

**Comment by amitaibu** [comment]
@niryariv ready for review.

File: app/_posts/2016-01-07-apps-entity-restrictions.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+permalink: "/content/apps-entity-restrictions"
+author: RoySegall
+layout: post
+published: true
+---
+
+
+{% include setup %}
+
+You might have heard of [Burning Man](http://burningman.org/). Basically it's a lot of hippies settling down in the desert for a few days, setting up small camps with diffrent themes that make up a big, temporary city.
```
Comment (niryariv) [inline app/_posts/2016-01-07-apps-entity-restrictions.md]: Add a photo, rarely we have subject matters with such photogenic potential

File: app/_posts/2016-01-07-apps-entity-restrictions.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+<div class="thumbnail">
+    <iframe src="http://gfycat.com/ifr/FragrantUnequaledHerculesbeetle" frameborder="0" scrolling="no" width="800" height="600" style="-webkit-backface-visibility: hidden;-webkit-transform: scale(1);" ></iframe>
+  <div class="caption">Apps Entity Restrictions in action</div>
+</div>
+
+[Apps Entity Restrictions](https://github.com/RoySegall/apps_entity_restrictions) is now our answer to the problem.
+
+<!-- more -->
+
+With this module suite you can create 3rd party application representation where you can determine which field or property each registered application can access on each entity. You can even restrict the allowed CRUD operations.
```
Comment (niryariv) [inline app/_posts/2016-01-07-apps-entity-restrictions.md]: Did you create this module or are you explaining to use a module someone else made? If you created, mention that

File: app/_posts/2016-01-07-apps-entity-restrictions.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+While working on Gizra's modules and projects in the past years I came to realize that a good API and a good DX is the one thing responsible for a good module. By default, any app is restricted from doing any operation - you know, security.
```
Comment (niryariv) [inline app/_posts/2016-01-07-apps-entity-restrictions.md]: What's DX? I'm assuming "developer experience" but took me a few minutes to realize.. just write the actual words.

File: app/_posts/2016-01-07-apps-entity-restrictions.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+{% include setup %}
+
+You might have heard of [Burning Man](http://burningman.org/). Basically it's a lot of hippies settling down in the desert for a few days, setting up small camps with diffrent themes that make up a big, temporary city.
+
+It's not for me.
+
+And yet, I found myself going to the hackathon of our local Burning man community as an enthusiastic Drupal developer willing to solve any Drupal issue (and you can assume they had a few). My part was to write the backend, and the obvious choice was using the [RESTful](https://github.com/RESTful-Drupal/restful) module.
+
+Soon I came across a big problem: How can we manage 3rd party applications and make sure they can't access resources which they shouldn't have access to?  
+How can we prevent the Secret Santa application (an app that provides addresses of other Burning Man attendees so they could receive gifts) from accessing a user's medical qualification data?
```
Comment (niryariv) [inline app/_posts/2016-01-07-apps-entity-restrictions.md]: You're talking about an architecture where Drupal is a backend serving several unrelated front end apps, some of whom are mobile apps, some web apps etc. This is some cool shit, expand on it: what apps are there, where they run, show some chart with a big box "Drupal" and small boxes for the apps and arrows between them, etc

File: app/_posts/2016-01-07-apps-entity-restrictions.md (hunk 5 above)
Comment (amitaibu) [inline app/_posts/2016-01-07-apps-entity-restrictions.md]: Like UI, we assume readers know it

Comment (amitaibu) [inline app/_posts/2016-01-07-apps-entity-restrictions.md]: (the post is intended for intermidate-advanced audience)

File: app/_posts/2016-01-07-apps-entity-restrictions.md (hunk 6 above)
Comment (amitaibu) [inline app/_posts/2016-01-07-apps-entity-restrictions.md]: > show some chart with a big box "Drupal" and small boxes for the apps and arrows between them, etc

I don't think it's needed. The concept of headless Drupal is already becoming known, and the audience here is assumed to know about it already

Comment (niryariv) [inline app/_posts/2016-01-07-apps-entity-restrictions.md]: not referring to headless, but to the model where a single drupal backend powers a bunch of separate, unrelated apps (eg a gifts app, an visitor registration app, a medical app - all independent of each other). i assumed this is fairly novel and is the main reason for entity restriction.

**Comment by RoySegall** [comment]
@amitaibu @niryariv Ready for review.

File: app/_posts/2016-01-14-apps-entity-restrictions.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+{% include setup %}
+
+You might have heard of [Burning Man](http://burningman.org/). Basically it's a lot of hippies settling down in the desert for a few days, setting up small camps with diffrent themes that make up a big, temporary city.
+
+It's not for me.
+
+<div class="thumbnail">
+  <img src="{{BASE_PATH}}/assets/images/posts/apps-entity-restrictions/burner.jpg">
+  <div class="caption">Radical Self-expression is one of <a href="http://midburn.org/en-ten-principles/">Midburn ten principles</a>. Copy rights reserved for Eyal levkovich.</div>
```
Comment (niryariv) [inline app/_posts/2016-01-14-apps-entity-restrictions.md]: Copy rights reserved for Eyal levkovich => (c) Eyal Levkovich

Comment (amitaibu) [inline app/_posts/2016-01-14-apps-entity-restrictions.md]: I'll fix

**Comment by RoySegall** [comment]
@amitaibu I returned the copy right section.

**Comment by amitaibu** [comment]
Merged, thanks!

http://www.gizra.com/content/apps-entity-restrictions/

**Pull request closed without merging**